src/main/java/com/salesforce/mirus/HerderStatusMonitor.java

```diff
@@ -91,42 +90,50 @@ private void onConnectors(Throwable error, Collection<String> connectorNames) {
 
   private void processConnector(String connectorName) {
     ConnectorStateInfo stateInfo = herder.connectorStatus(connectorName);
-    if (workerId.equals(stateInfo.connector().workerId())) {
-      if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
-        connectorJmxReport.incrementConnectorRestartAttempts(connectorName);
-        if (autoRestartConnectorEnabled) {
+
+    if (autoRestartConnectorEnabled) {
+      if (workerId.equals(stateInfo.connector().workerId())) {
+        if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
```

Comment:
Also, what do you think about collapsing these three if statements into a single condition? We only do anything if all three are true.